client/src/providers/UserProvider.tsx

```diff
@@ -42,10 +47,14 @@ export function UserProvider(props: React.PropsWithChildren<{}>): JSX.Element {
   ): Promise<void> {
     let newUser: User | null = user;
     if (shouldFetch) {
+      setIsLoading(true);
```

Comment:
feel free to test this theory by changing the setTimeout timer to like 10 seconds or something...

the way this works is it automatically makes a request to get a new token right before the token is about to expire (1hr)
at that time i think this would cause our page to show the "loading" text you've added in `PrivateRouteWrapper/index.tsx`
imo the user won't want to see the loading text flash in the middle of their session so i think we should only set this isLoading on our initial fetch
i think wrapping this setter in an `if newUser === null` will work

Reply:
Nice catch, I figured out that this line doesn't actually need to be there at all since the loading state is defaulted to true

Reply:
even better!